### ISSUE TITLE
update_index interfering with other after_commits

### ISSUE THREAD

**Issue by mlchai** [toptal/chewy#182]
I'm having a very odd problem where update_index is blocking other after_commit callbacks from firing. Moving the update_index around causes it or the other ones to fire, but I really need both of them to fire. I tried setting the use_after_commit_callbacks option but that didn't seem to do anything. Here's an example:

update_index
after_commit :test1

in this case, the update_index will not fire but test1 will. If I do

after_commit :test1
update_index

then the update_index will fire but test1 won't.

I know it's old, but it's very similar to behavior seen here:

https://github.com/rails/rails/issues/988
http://apidock.com/rails/ActiveRecord/Transactions/ClassMethods/after_commit

Is there a compelling reason why after_commit is the default rather than after_save?

**Comment by pyromaniac** [comment]
use_after_commit_callbacks should be set before models loading, e.g. in initializers
The reason is: we have to guarantee that data in in database already while indexing.

Didn't face this problem before, needs debugging here

**Comment by caldwecr** [comment]
I agree that it needs to remain on after commit as pyro explained. This is a pretty common issue of late though.

**Comment by mlchai** [comment]
I'm already using the use_after_commit_callbacks in an initializer. If I keep it as after_commit, does that mean I can't use any other after_commits in my application?

**Comment by pyromaniac** [comment]
Don't know, it looks like a rails bug, so we just need to find a solution

**Comment by mlchai** [comment]
Just an update, I found out what was wrong. I wasn't defining a strategy before, so it was throwing the undefined strategy exception but since it's an after_commit the exception was being swallowed. I had no idea, and it seems a side effect of this is that when an exception is thrown all other after_commits won't be run. Definitely a rails bug. Closing the issue here since this isn't something to do with Chewy.
